### PR TITLE
🎨 Improve slicer to phpunit9

### PR DESCRIPTION
**Pull request by casahugo**
Hey Guys ! fix https://github.com/wizaplace/phpunit-slicer/issues/7

- You can no longer extend the TestRunner. The slice is done before launching the runner

- Management of the CI in github action, just activate the option

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: CI
+
+on: [push]
+
+jobs:
+  build-test:
+    runs-on: ubuntu-latest
```
Comment (philippe-vandermoere) [inline .github/workflows/ci.yml:7]:
```suggestion
    runs-on: ubuntu-latest
    strategy:
      php-versions: ['7.1', '7.2', '7.3', '7.4']
    name: PHP ${{ matrix.php-versions }}
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+    steps:
+      - uses: actions/checkout@v1
+      - uses: php-actions/composer@v1 # or alternative dependency management
+      - name: phpunit
+        run: |
+          sudo vendor/bin/phpunit
```
Comment (philippe-vandermoere) [inline .github/workflows/ci.yml:14]:
```suggestion
      - name: Checkout
        uses: actions/checkout@v1
      - name: Setup PHP
        uses: shivammathur/setup-php@v2
        with:
          php-version: ${{ matrix.php-versions }}
          tools: composer:v1
      - name: composer
        run: composer install --no-interaction
      - name: phpunit
        run: vendor/bin/phpunit
```

File: composer.json [hunk 3]
```diff
@@ -3,11 +3,12 @@
...
     "require": {
         "php": "^7.1",
-        "phpunit/phpunit": "^7|^8"
+        "phpunit/phpunit": "^7|^8|^9"
     },
     "autoload": {
         "psr-4": {
-            "Wizaplace\\PHPUnit\\Slicer\\": "src/"
+            "Wizaplace\\PHPUnit\\Slicer\\": "src/",
+            "Wizaplace\\PHPUnit\\Tests\\Slicer\\": "tests/"
```
Comment (philippe-vandermoere) [inline composer.json:11]: Move "Wizaplace\PHPUnit\\Tests\Slicer" autoloader to the section "autoload-dev"

**Review by philippe-vandermoere — CHANGES_REQUESTED** [review]

**Review by philippe-vandermoere — APPROVED** [review]

**Comment by mfn** [comment]
Unfortunately I cannot confirm this to work?

I had to hack around to get this installed in my project as I'm on PHPUnit 9.5, so I added a repository to my `composer.json` like this, pointing specifically to the last commit in this PR; this was necessary because the master branch `composer.json` requirements don't allow this to be installable for PHPUnit > 8:
```json
  "repositories": [
    {
      "type": "package",
      "package": {
        "name": "wizaplace/phpunit-slicer",
        "type": "library",
        "version": "1234",
        "dist": {
          "url": "https://codeload.github.com/wizaplace/phpunit-slicer/legacy.zip/5b13677645c81689a5f5dd31ac88fa3240b741c7",
          "type": "zip"
        },
        "autoload": {
          "psr-4": {
            "Wizaplace\\PHPUnit\\Slicer\\": "src/"
          }
        },
        "bin": [
          "phpunit-slicer"
        ]
      }
    }
  ],
```
I.e. the point is to not have a `require` temporarily to solve the dependency resolution. Also version `1234` is just random 😏 

I've then installed it via `composer require --dev wizaplace/phpunit-slicer:1234 --prefer-dist`

Anyhow, I then tried this in my project:
```
$ vendor/bin/phpunit-slicer --slice 1/6
PHPUnit suite slicer, running slice 1/6 (2267 tests: from #1 to #2267)
PHPUnit 9.5.0 by Sebastian Bergmann and contributors.

No tests executed!
```
or
```
$ vendor/bin/phpunit-slicer --slice 2/6
PHPUnit suite slicer, running slice 2/6 (2267 tests: from #2268 to #4534)
PHPUnit 9.5.0 by Sebastian Bergmann and contributors.

No tests executed!
```
😢 

Regular phpunit works:
```
$ vendor/bin/phpunit
PHPUnit 9.5.0 by Sebastian Bergmann and contributors.

..............^C
```

There's not much more "magic" going on and a previous hack version of phpunit-slicer works too (i.e. the copypasta approach from https://github.com/wizaplace/phpunit-slicer/pull/8

Any ideas?

**Comment by mfn** [comment]
Here's the `phphunit.xml.dist` I'm using:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"
         backupGlobals="false"
         backupStaticAttributes="false"
         bootstrap="tests/bootstrap.php"
         colors="true"
         convertErrorsToExceptions="true"
         convertNoticesToExceptions="true"
         convertWarningsToExceptions="true"
         processIsolation="false"
         stopOnFailure="false"
         cacheResult="false"
         failOnWarning="true"
>
  <testsuites>
    <testsuite name="Application Test Suite">
      <directory>./tests/</directory>
    </testsuite>
  </testsuites>
  <coverage>
    <include>
      <directory suffix=".php">app/</directory>
    </include>
  </coverage>
</phpunit>
```
And here the `tests/bootstrap.php`, just useful Laravel-ish thing added befiore the regular vendor/autoload.php; I also tried without them but no change:
```php
<?php declare(strict_types = 1);

// Always ensure the cached bootstrap files are removed before running tests
// Especially important for auto-discover installed packages
// when switching branches
foreach (glob(__DIR__ . '/../bootstrap/cache/*.php') as $filename) {
    unlink($filename);
}

require __DIR__ . '/../vendor/autoload.php';
```

**Pull request closed without merging**